src/gromacs/nblib/gmxsetup.h

```diff
@@ -67,17 +67,32 @@ enum class CombinationRule : int
     Count     = 1
 };
 
-struct NbvSetupUtil
+class NbvSetupUtil
 {
+public:
     NbvSetupUtil(SimulationState system, const NBKernelOptions& options);
 
+    //! Sets up and returns a GmxForceCalculator
+    std::unique_ptr<GmxForceCalculator> setupGmxForceCalculator();
+
+private:
+    //! Reads Topology data from the user-facing layer to translate to GMX data structures
     void unpackTopologyToGmx();
 
-    std::unique_ptr<nonbonded_verlet_t> setupNbnxmInstance();
+    //! Puts particles on a grid based on bounds specified by the box
+    void setParticlesOnGrid(std::unique_ptr<nonbonded_verlet_t>& nbv);
 
-    std::unique_ptr<GmxForceCalculator> setupGmxForceCalculator();
+    //! Sets Particle Types and Charges and VdW params
+    void setAtomProperties(std::unique_ptr<nonbonded_verlet_t>& nbv);
 
+    //! Sets up non-bonded verlet on the GmxFOrceCalculator
+    std::unique_ptr<nonbonded_verlet_t> setupNbnxmInstance();
+
+public:
```

Comment:
These data members are not being used outside the class.
So, I vote that they actually disappear and get passed as arguments to the functions that need them...

Reply:
Yes. Working on this.